packages/dashboard/webpack.config.ts

```diff
@@ -0,0 +1,103 @@
+import * as webpack from "webpack";
+import type WebpackDevServer from "webpack-dev-server";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import EslintWebpackPlugin from "eslint-webpack-plugin";
+import TerserWebpackPlugin from "terser-webpack-plugin";
+import path from "path";
+
+const isProduction = process.env.NODE_ENV === "production";
+
+const progressHandler = (function () {
+  let lastLoggedPercent = -1;
+  const minInterval = 15;
+
+  return {
+    log(percent: number, _message: string, ..._args: string[]) {
+      percent = Math.round(percent * 1000) / 10;
+      if (lastLoggedPercent < 0 || percent - lastLoggedPercent >= minInterval) {
+        lastLoggedPercent = percent;
+        console.log(`Compiling: ${percent}%`);
+      } else if (percent === 100) {
+        console.log(`Finished compiling`);
+      }
+    }
+  };
+})();
+
+const config: webpack.Configuration = {
+  mode: isProduction ? "production" : "development",
+  entry: "./src/index.tsx",
+  output: {
+    filename: "bundle.js",
+    path: path.resolve("build")
+  },
+  optimization: {
+    minimize: isProduction,
+    minimizer: [
+      new TerserWebpackPlugin({
+        terserOptions: {
+          compress: {
+            drop_console: isProduction
+          }
+        }
+      })
+    ]
+  },
+  resolve: {
+    extensions: [".ts", ".tsx", ".js", ".jsx"],
+    alias: {
+      src: path.resolve("src")
+    },
+    fallback: {
+      crypto: require.resolve("crypto-browserify"),
+      os: require.resolve("os-browserify/browser"),
+      path: require.resolve("path-browserify"),
+      stream: require.resolve("stream-browserify"),
+      vm: require.resolve("vm-browserify")
+    }
+  },
+  devtool: "source-map",
+  devServer: {
+    port: 3000,
+    historyApiFallback: true,
+    client: {
+      reconnect: false,
+      overlay: {
+        errors: true,
+        warnings: false
+      }
+    }
+  },
+  stats: "minimal",
+  module: {
+    rules: [
+      {
+        test: [/\.jsx?$/, /\.tsx?$/],
+        use: ["babel-loader"],
+        exclude: /node_modules/
+      },
```

Comment:
I think babel-loader is maybe fine, but for consistency with the work that I'm doing, you might want to consider going with ts-loader.

That said, babel loader doesn't check types at all, so unless you add a separate build step to check for type errors, CI will be completely blind to any issues there.

Reply:
I certainly wasn't aware of the nuances (babel just throws the types away?? I was fooled by my assumption about [this plugin](https://babeljs.io/docs/en/babel-preset-typescript)). I've [updated](https://github.com/trufflesuite/truffle/commit/2f898673baff2df7dfe63c2893b13d75eadcf64e) it to ts-loader to align with your stuff (and you know, to check types).

Reply:
Thanks heaps - that'll make my life a lot easier for the remainder of the typescript support/bundling stuff!